### PR TITLE
Error Checking Service File Reload

### PR DESCRIPTION
**Pull request by MiaAltieri**
### Prelude 
Quick Tiny PR for a point @delgod made on a [previous PR](https://github.com/canonical/mongodb-operator/pull/36#discussion_r873593533). 

_Note_ after discussion with @delgod and @WRFitch we decided it best to error check for systemd, but allow permissions to raise exceptions.

### Problem

Reloading service file didn't have any error checking

### Solution

solution add error checking for reloading systemd 

### Testing

Added a unit test to verify behavior in case of failure.

### CODE REVIEW

**Review by paulomach — APPROVED** [review]
LGTM

File: src/charm.py [hunk 1]
```diff
@@ -239,7 +239,12 @@ def _on_install(self, event) -> None:
         os.chown(MONGO_DATA_DIR, mongodb_user.pw_uid, mongodb_user.pw_gid)
 
         # changes to service files are only applied after reloading
-        systemd.daemon_reload()
+        try:
+            systemd.daemon_reload()
+        except systemd.SystemdError:
+            logger.error("failed to install MongoDB")
+            self.unit.status = BlockedStatus("couldn't start MongoDB")
+            return
```
[the remark this answers is not in the document]
Comment (marcoppenheimer) [reply src/charm.py:247]: Yeah I'd `raise` instead of return here, as there's the likelihood is that a user will have to fix it.

**Review by marcoppenheimer — CHANGES_REQUESTED** [review]

File: src/charm.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (MiaAltieri) [reply src/charm.py:247]: Great point! Let me ask the Charming Newbies channel!

[the remark this answers is not in the document]
Comment (marcoppenheimer) [reply src/charm.py:247]: I'd suggest the following:

```python
except systemd.SystemdError as e:
    self.unit.status = BlockedStatus("failed to reload mongo daemon") # or equivalent
    raise e
```

Raise will give full information in the debug logs.
Status message gives top-line user information

logger.error is not needed

**Comment by MiaAltieri** [comment]
Closing this pull request as failures in SystemD indicate a greater issue which is out of scope for what this charm should handle.

Further the suggestion of raised exceptions overwrite set status messages and in the solution:
```
        except systemd.SystemdError as e:
            logger.error("failed to reload /etc/systemd/system/mongod.service file, error: %s", e)
            self.unit.status = BlockedStatus("failed to reload MongoDB service")
            raise
```
 `BlockedStatus` gets overwritten by `raise`, and hence becomes unused. It could be changed it to:
```
        except systemd.SystemdError as e:
            logger.error("failed to reload /etc/systemd/system/mongod.service file, error: %s", e)
            raise
```
but this is excessive logging. It is best to allow `systemd.daemon_reload()` to error out

**Pull request closed without merging**